pkg/apimachinery/registered/registered.go

```diff
@@ -59,15 +69,27 @@ func init() {
 	}
 }
 
-// RegisterVersions add the versions the registeredVersions.
-func RegisterVersions(versions ...unversioned.GroupVersion) {
-	for _, v := range versions {
+// RegisterVersions adds the given group versions to the list of registered group versions.
+func RegisterVersions(availableVersions []unversioned.GroupVersion) {
+	for _, v := range availableVersions {
 		registeredVersions[v] = struct{}{}
 	}
 }
 
-// EnableVersions add the versions to the enabledVersions. The caller of this
-// function is responsible to add the version to 'latest' and 'Scheme'.
+// RegisterGroup adds the given group to the list of registered groups.
+func RegisterGroup(groupMeta apimachinery.GroupMeta) error {
```

Comment:
I am planning to merge RegisterGroup and RegisterVersion into one.
GroupMeta has all the required information.

It requires moving enableVersion logic as well (since we cannot create GroupMeta and RestMapper unless objects have been added to schema), which I did not do in this PR.
Will send another PR for it.